### DISCUSSION TITLE
How to partial-compile

### DISCUSSION THREAD

**Discussion by TornaxO7** [bilelmoussaoui/flatpak-vscode#191] [Q&A]
Hello! Neovim-User here :)
May I ask you how this extension is able to partially-compile a rust project? Because all I could find with `flatpak-builder` is to remove the `flatpak-app` directory (or use the `--force-clear` argument) which rebuilds *everything* but someone told me that the vscode-extension can rebuild the app without recompiling *everything* again.

**Comment by bilelmoussaoui** [comment]
The extension uses flatpak-builder to compile all the dependencies (modules) except the last one, then spawns meson/cmake/other build system manually using flatpak itself inside the build repository. This way you would build the current tree instead of whatever the flatpak manifest points to.

You could look at the source code to get an idea of how to do that, or use something like fenv https://gitlab.gnome.org/ZanderBrown/fenv
